src/index.js

```diff
@@ -187,7 +187,7 @@ export default class ServiceWorkerPlugin {
       const serviceWorkerOptionInline = JSON.stringify(
         serviceWorkerOption,
         null,
-        this.options.minimize ? 0 : 2
+        (compiler.options.mode === 'production' || this.options.minimize) ? 0 : 2
```

Comment:
If minimize is set explicitly as an option, the compiler mode "production" shouldn't override this. Could you change this to only consider compiler mode, if this.options.minimize is undefined?

Reply:
Sure, I'll check it out!